### PR TITLE
Changes to configuration

### PR DESCRIPTION
**Pull request by Xpl0itR**
- Generates config.json from default values if config.json is not present
- Load config from either ryufs or ryujinx.exe parent folder with ryujinx.exe parent folder taking priority

### CODE REVIEW

**Comment by jduncanator** [comment]
Can you PR this again without all the formatting changes? You may need to disable your ReSharper before doing so. Thanks.

**Comment by Thog** [comment]
Blocked by #812 .

**Comment by Xpl0itR** [comment]
rebased, ready for review

**Comment by Xpl0itR** [comment]
done

File: Ryujinx/Program.cs [hunk 1]
```diff
@@ -28,19 +31,33 @@ static void Main(string[] args)
             // Initialize Discord integration
             DiscordIntegrationModule.Initialize();
 
-            string configurationPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Config.json");
+            string localConfigurationPath  = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Config.json");
+            string globalConfigurationPath = Path.Combine(new VirtualFileSystem().GetBasePath(), "Config.json");
```
Comment (gdkchan) [inline Ryujinx/Program.cs]: `GetBasePath` should be static, theres no reason to create a instance of VirtualFileSystem just for that. Although thats a existing problem so I'll approve.

**Pull request merged**